apps/condo/domains/acquiring/schema/CalculateFeeForReceiptService.js

```diff
@@ -0,0 +1,124 @@
+const { isNil } = require('lodash')
+
+const { GQLError } = require('@open-condo/keystone/errors')
+const { GQLCustomSchema, getById, find } = require('@open-condo/keystone/schema')
+
+const access = require('@condo/domains/acquiring/access/CalculateFeeForReceiptService')
+const {
+    RESIDENT_CALCULATE_FEE_FOR_RECEIPT_WINDOW_IN_SEC,
+    MAX_RESIDENT_CALCULATE_FEE_FOR_RECEIPT_WINDOW_IN_SEC_CALLS_BY_WINDOW_SEC,
+} = require('@condo/domains/acquiring/constants/constants')
+const { CONTEXT_FINISHED_STATUS } = require('@condo/domains/acquiring/constants/context')
+const { GQL_ERRORS: { PAYMENT_AMOUNT_LESS_THAN_MINIMUM } } = require('@condo/domains/acquiring/constants/errors')
+const {
+    getAcquiringIntegrationContextFormula,
+    FeeDistribution,
+} = require('@condo/domains/acquiring/utils/serverSchema/feeDistribution')
+const { RESIDENT } = require('@condo/domains/user/constants/common')
+const { RedisGuard } = require('@condo/domains/user/utils/serverSchema/guards')
+const ERRORS = {
+    PAYMENT_AMOUNT_LESS_THAN_MINIMUM: {
+        ...PAYMENT_AMOUNT_LESS_THAN_MINIMUM,
+        query: 'calculateFeeForReceipt',
+    },
+}
+
+const redisGuard = new RedisGuard()
+
+const checkLimits = async (userId) => {
+    await redisGuard.checkCustomLimitCounters(
+        `calculate_fee_for_receipt:user:${userId}`,
+        RESIDENT_CALCULATE_FEE_FOR_RECEIPT_WINDOW_IN_SEC,
+        MAX_RESIDENT_CALCULATE_FEE_FOR_RECEIPT_WINDOW_IN_SEC_CALLS_BY_WINDOW_SEC,
+    )
+}
+
+const CalculateFeeForReceiptService = new GQLCustomSchema('CalculateFeeForReceiptService', {
+    types: [
+        {
+            access: true,
+            type: 'input CalculateFeeForReceiptInput { receipt: BillingReceiptWhereUniqueInput!, amount: String! }',
+        },
+        {
+            access: true,
+            type: 'type CalculateFeeForReceiptOutput { amountWithoutExplicitFee: String!, explicitFee: String!, explicitServiceCharge: String! }',
+        },
+    ],
+    
+    queries: [
+        {
+            access: access.canCalculateFeeForReceipt,
+            schema: 'calculateFeeForReceipt (data: CalculateFeeForReceiptInput!): CalculateFeeForReceiptOutput',
+            resolver: async (parent, args, context) => {
+                if (context.authedItem.type === RESIDENT) {
+                    await checkLimits(context.authedItem.id)
+                }
+
+                const { data: { receipt: { id: receiptId }, amount } } = args
+
+                const [billingReceipt] = await find('BillingReceipt', {
+                    id: receiptId,
+                    deletedAt: null,
+                    context: {
+                        deletedAt: null,
+                        integration: {
+                            deletedAt: null,
+                        },
+                    },
+                })
+
+                if (isNil(billingReceipt)) {
+                    throw new GQLError({
+                        ...ERRORS.CANNOT_FIND_BILLING_RECEIPT,
```

Comment:
Where this error defined?
<img width="1193" alt="image" src="https://github.com/user-attachments/assets/497effde-5c72-49d9-97da-ab8a42bbbf91">